### PR TITLE
`sage.graphs`: Update `# needs`, use block-scoped tags

### PR DESCRIPTION
**Pull request by mkoeppe**
Cherry-picked from
- #35095
Part of
- #29705

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/graphs/connectivity.pyx [hunk 1]
```diff
@@ -1441,13 +1441,13 @@ def vertex_connectivity(G, value_only=True, sets=False, k=None, solver=None, ver
        sage: from sage.graphs.connectivity import is_strongly_connected
        sage: from sage.graphs.connectivity import is_connected
        sage: empty = Graph()
-       sage: vertex_connectivity(empty)
+       sage: vertex_connectivity(empty)                                                 # needs sage.numerical.mip
```
Comment (dcoudert) [inline src/sage/graphs/connectivity.pyx:1444]: could be at block level

File: src/sage/graphs/generators/families.py [hunk 2]
```diff
@@ -2680,13 +2693,14 @@ def SwitchedSquaredSkewHadamardMatrixGraph(n):
 
     EXAMPLES::
 
-        sage: g=graphs.SwitchedSquaredSkewHadamardMatrixGraph(4)
-        sage: g.is_strongly_regular(parameters=True)
+        sage: g = graphs.SwitchedSquaredSkewHadamardMatrixGraph(4)                      # needs sage.modules
```
Comment (dcoudert) [inline src/sage/graphs/generators/families.py:2696]: should be block level

**Review by dcoudert — APPROVED** [review]
There are few places where block tags would certainly be nicer, but it's a matter of preferences.

Let's hope this giant PR will not cause too many conflicts with other open PR on graphs...

**Comment by mkoeppe** [comment]
Thank you!

**Comment by vbraun** [comment]
There is a :hankey: in src/sage/graphs/generators/smallgraphs.py which is probably also why the CI fails. Can we avoid settings positive review in that case?

**Comment by mkoeppe** [comment]
Sorry about this.

Probably we should have a linter that checks for conflict markers.
But then someone should make running the linter part of their merging workflow too...

**Pull request merged**